src/couch_epi_codegen.erl

```diff
@@ -0,0 +1,72 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License. You may obtain a copy of
+% the License at
+%
+% http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(couch_epi_codegen).
+
+-export([generate/2, scan/1, parse/1, function/1, format_term/1]).
+
+generate(ModuleName, Forms) when is_atom(ModuleName) ->
+    generate(atom_to_list(ModuleName), Forms);
+generate(ModuleName, Forms0) ->
+    Forms = scan("-module(" ++ ModuleName ++ ").") ++ Forms0,
+    ASTForms = parse(Forms),
+    {ok, Mod, Bin} = compile:forms(ASTForms, [verbose, report_errors]),
+    {module, Mod} = code:load_binary(Mod, atom_to_list(Mod) ++ ".erl", Bin),
+    ok.
+
+scan(String) ->
+    Exprs = [E || E <- re:split(String, "\\.\n", [{return, list}, trim])],
+    FormsTokens = lists:foldl(fun(Expr, Acc) ->
+        case erl_scan:string(Expr) of
+            {ok, [], _} ->
+                Acc;
+            {ok, Tokens, _} ->
+                [{Expr, fixup_terminator(Tokens)} | Acc]
+        end
+    end, [], Exprs),
+    lists:reverse(FormsTokens).
+
+parse(FormsTokens) ->
+    ASTForms = lists:foldl(fun(Tokens, Forms) ->
+        {ok, AST} = parse_form(Tokens),
+        [AST | Forms]
+    end, [], FormsTokens),
+    lists:reverse(ASTForms).
+
+format_term(Data) ->
+    lists:flatten(io_lib:format("~p", [Data])).
```

Comment:
Right. I'm actually considering wrapping Data with `term_to_binary` to make sure it works even if things like `pid()` or `ref()` are passed by accident.